ModuleDescriptionParser/ModuleParameter.h

```diff
@@ -78,6 +79,27 @@ class ModuleDescriptionParser_EXPORT ModuleParameter
     return this->Reference;
   }
 
+  virtual void AddForwardReference(const std::string &role, const std::string &ref)
+  {
+    std::map<std::string,std::vector<std::string> >::iterator frit =
+      this->ForwardReferences.find(role);
+    if (frit != this->ForwardReferences.end())
+      {
+      this->ForwardReferences[role].push_back(ref);
+      }
+    else
+      {
+      std::vector<std::string> refVector(1, ref);
+      this->ForwardReferences[role] = refVector;
+      }
+  }
+
+  virtual const void GetForwardReferences(
```

Comment:
@cpinter  New warnings encountered from your PR.  

In file included from /nfsscratch/opt/iiai/apps/BRAINSTools/20190321/BRAINSTools-bld/SlicerExecutionModel/ModuleDescriptionParser/ModuleParameterGroup.h:19:0,
                 from /nfsscratch/opt/iiai/apps/BRAINSTools/20190321/BRAINSTools-bld/SlicerExecutionModel/ModuleDescriptionParser/ModuleParameterGroup.cxx:15:
/nfsscratch/opt/iiai/apps/BRAINSTools/20190321/BRAINSTools-bld/SlicerExecutionModel/ModuleDescriptionParser/ModuleParameter.h:97:11: warning: type qualifiers ignored on function return type [-Wignored-qualifiers]
   virtual const void GetForwardReferences(
           ^~~~~
In file included from /nfsscratch/opt/iiai/apps/BRAINSTools/20190321/BRAINSTools-bld/SlicerExecutionModel/ModuleDescriptionParser/ModuleParameter.cxx:14:0:
/nfsscratch/opt/iiai/apps/BRAINSTools/20190321/BRAINSTools-bld/SlicerExecutionModel/ModuleDescriptionParser/ModuleParameter.h:97:11: warning: type qualifiers ignored on function return type [-Wignored-qualifiers]
   virtual const void GetForwardReferences(
           ^~~~~
In file included from /nfsscratch/opt/iiai/apps/BRAINSTools/20190321/BRAINSTools-bld/SlicerExecutionModel/ModuleDescriptionParser/ModuleParameterGroup.h:19:0,
                 from /nfsscratch/opt/iiai/apps/BRAINSTools/20190321/BRAINSTools-bld/SlicerExecutionModel/ModuleDescriptionParser/ModuleDescription.h:20,
                 from /nfsscratch/opt/iiai/apps/BRAINSTools/20190321/BRAINSTools-bld/SlicerExecutionModel/ModuleDescriptionParser/ModuleDescriptionParser.cxx:16:
/nfsscratch/opt/iiai/apps/BRAINSTools/20190321/BRAINSTools-bld/SlicerExecutionModel/ModuleDescriptionParser/ModuleParameter.h:97:11: warning: type qualifiers ignored on function return type [-Wignored-qualifiers]
   virtual const void GetForwardReferences(
           ^~~~~
In file included from /nfsscratch/opt/iiai/apps/BRAINSTools/20190321/BRAINSTools-bld/SlicerExecutionModel/ModuleDescriptionParser/ModuleParameterGroup.h:19:0,
                 from /nfsscratch/opt/iiai/apps/BRAINSTools/20190321/BRAINSTools-bld/SlicerExecutionModel/ModuleDescriptionParser/ModuleDescription.h:20,
                 from /nfsscratch/opt/iiai/apps/BRAINSTools/20190321/BRAINSTools-bld/SlicerExecutionModel/ModuleDescriptionParser/BatchMakeUtilities.h:20,
                 from /nfsscratch/opt/iiai/apps/BRAINSTools/20190321/BRAINSTools-bld/SlicerExecutionModel/ModuleDescriptionParser/BatchMakeUtilities.cxx:1:
/nfsscratch/opt/iiai/apps/BRAINSTools/20190321/BRAINSTools-bld/SlicerExecutionModel/ModuleDescriptionParser/ModuleParameter.h:97:11: warning: type qualifiers ignored on function return type [-Wignored-qualifiers]
   virtual const void GetForwardReferences(
           ^~~~~
In file included from /nfsscratch/opt/iiai/apps/BRAINSTools/20190321/BRAINSTools-bld/SlicerExecutionModel/ModuleDescriptionParser/ModuleParameterGroup.h:19:0,
                 from /nfsscratch/opt/iiai/apps/BRAINSTools/20190321/BRAINSTools-bld/SlicerExecutionModel/ModuleDescriptionParser/ModuleDescription.h:20,
                 from /nfsscratch/opt/iiai/apps/BRAINSTools/20190321/BRAINSTools-bld/SlicerExecutionModel/ModuleDescriptionParser/ModuleDescription.cxx:15:
/nfsscratch/opt/iiai/apps/BRAINSTools/20190321/BRAINSTools-bld/SlicerExecutionModel/ModuleDescriptionParser/ModuleParameter.h:97:11: warning: type qualifiers ignored on function return type [-Wignored-qualifiers]
   virtual const void GetForwardReferences(
           ^~~~~

Reply:
Thanks for letting me know. I did't get these warnings with Visual Studio. I'm fixing this (as I cannot test it I can just guess that the problem is the const before the void, which was not intended of course).

I get 357 other warnings though, like this:
1>c:\d\s4d\slicerexecutionmodel\moduledescriptionparser\ModuleParameter.h(399): warning C4251: 'ModuleParameter::Name': class 'std::basic_string<char,std::char_traits<char>,std::allocator<char>>' needs to have dll-interface to be used by clients of class 'ModuleParameter'
Do they occur on linux as well?